api/app/server.py

```diff
@@ -0,0 +1,59 @@
+from fastapi import FastAPI
+from fastapi.responses import RedirectResponse
+from fastapi.middleware.cors import CORSMiddleware
+
+from langserve import add_routes
+from langchain_core.runnables import RunnableConfig
+
+from app.utils.events import lifespan
+from rag.chains import rag_chain_with_source
+from rag.utils.callbacks import langfuse_handler_from_config
+
+from app.client import chat
+
+
+app = FastAPI(lifespan=lifespan)
+
+# CORS
+origins = ["http://localhost:8000", "http://localhost"]
+
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=origins,
+    allow_credentials=True,
+    allow_methods=["POST"],  # TODO: change to ["POST", "GET"] when ready
+    allow_headers=["*"],
+)
+
+langfuse_handler = langfuse_handler_from_config(
+    trace_name="hp_installer_chatbot_chain",
+    user_id="testing",
+    session_id="local",
+    version="0.1.0",
+    release="0.1.0",
+    tags=["dev", "api", "v1"],
+)  # TODO move to env vars
+
+app.add_route("/hook", chat, methods=["POST"])
+
+chatbot_chain = rag_chain_with_source.with_config(
```

Comment:
Not too sure what's this doing?

Reply:
currently for ease, although there's a separate "whatsapp interface" client.py file, the whatsapp client and the chatbot chains api scripts are hosted on the same instance.

This adds the /hook endpoint as a POST method.

The separate add_routes() methods a few lines below is a method from LangServe that abstracts away adding schema, and the extra endpoints and processing needed for /stream /batch and other endpoints.